### ISSUE TITLE
Build for ios fails : Undefined symbols for architecture armv7:

### ISSUE THREAD

**Issue by abaturyna** [tekartik/sqflite#536]
Command CompileSwift failed with a nonzero exit code
    Command CompileSwift failed with a nonzero exit code
    ld: warning: ignoring file /Users/nastya/Dropbox/BUSINESS/project/build/ios/Release-iphoneos/FMDB/FMDB.framework/FMDB, building for iOS-armv7 but attempting to link with file built for iOS-arm64
    Undefined symbols for architecture armv7:
      "_OBJC_CLASS_$_FMDatabaseQueue", referenced from:
          objc-class-ref in SqflitePlugin.o

Flutter doctor : 
[✓] Flutter (Channel dev, 1.24.0-3.0.pre, on Mac OS X 10.15.7 19H2 darwin-x64, locale ru-BY)
    • Flutter version 1.24.0-3.0.pre at /Users/nastya/Dropbox/BUSINESS/Soft/flutter
    • Framework revision 2783f8e2e1 (6 дней назад), 2020-10-22 09:36:06 -0700
    • Engine revision defa8be2b1
    • Dart version 2.11.0 (build 2.11.0-242.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/nastya/Library/Android/sdk
    • Platform android-29, build-tools 29.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.1, Build version 12A7403
    • CocoaPods version 1.10.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 47.1.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] IntelliJ IDEA Ultimate Edition (version 2020.1.1)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 45.1.3
    • Dart plugin version 201.7223.97

[✓] Connected device (2 available)
    • Web Server (web) • web-server • web-javascript • Flutter Tools
    • Chrome (web)     • chrome     • web-javascript • Google Chrome 86.0.4240.111

**Comment by egodefroy** [comment]
I had the same issue.

works when in Podfile I have 
  platform :ios, '9.0' 

but not when 
  platform :ios, '11.0'

**Comment by vanfresh** [comment]
+1 I am also having the same problem.  Defining platform :ios, '11.0' does not fix it.

**Comment by abaturyna** [comment]
@egodefroy for me not helped.

@vanfresh could you please put emoji reaction for my first initial message so that the issue was prioritized above all for sqflite developers.

**Comment by egodefroy** [comment]
@abaturyna maybe you can try to upgrade cocoapods to a newer version : sudo gem install cocoapods

**Comment by abaturyna** [comment]
@egodefroy I have upgraded my cocoapods (1.0.10 at the moment) - did not helped. Also, cleaned pod cache, regenerated XCode derived data - nothing helped.

**Comment by abaturyna** [comment]
@egodefroy in addition to below message, my XCode version is Version 12.1 (12A7403

**Comment by egodefroy** [comment]
@abaturyna probably an issue with sqflite link process

**Comment by alextekartik** [comment]
Sorry for the troubles you're having. The update to [FMDB 2.7.5](https://github.com/tekartik/sqflite/issues/528) in 1.3.2 could be the issue. Support for iOS 8 has been dropped so you might try adding/chaning in the pod file to ios 9 (`platform :ios, '9.0'`).

I cannot reproduce on a fresh project.
Does it happen when you create a new project and add sqflite?
Most of time I spend on iOS is also dealing with build issue. Recreating a project to match new flutter evolution is sometimes needed (you can compare the settings between a new project and your existing one).

**Comment by abaturyna** [comment]
@alextekartik  as i stated earlier, having 'platform :ios, '9.0' in Podfile does not help. I even need to have 'platform :ios, '14.0' because of another package. And even 14.0 does not resolve the issue! the same is set to XCode workspace settings for target and runner (14.0) - does not help. 

My project is existing one. Podfile is recreated (the latest one). 

My dependencies in pubspec.yaml are : 
dependencies:
  flutter:
    sdk: flutter
  camera:
  path_provider:
  path:
  sqflite: ^1.3.2+1
  csv: ^4.0.3
  flutter_exif_rotation: ^0.2.7
  cupertino_icons: ^1.0.0
  image_picker: ^0.6.7+12
  fuzzy: ^0.2.3
  gallery_saver: ^2.0.1
  string_splitter: ^0.1.0+5
  image: ^2.1.17
  #admob_flutter: ^1.0.1
  isolate_handler: ^0.3.1
  rate_my_app: ^0.7.2
  launch_review: ^2.0.0
  purchases_flutter: ^1.4.1
  firebase_ml_vision: ^0.9.7
  number_display: ^2.3.0
  custom_radio_button: ^0.0.3
  flutter_swiper: ^1.1.6
  flutter_spinkit: ^4.1.2+1
  circular_check_box: ^1.0.4
  font_awesome_flutter: ^8.8.1
  url_launcher: ^5.7.0
  intl: ^0.16.1

**Comment by abaturyna** [comment]
@alextekartik  as workaround i reverted to sqflite: ^1.3.0 and this works. But the original issue should be fixed. If build error on ios occurs due to fmdb version change - then it was not tested properly from sqflite team before release. Thanks.

**Comment by Saifallak** [comment]
also have the same problem, reverted back to `1.3.0` and it's not fixed too

**Comment by Saifallak** [comment]
had to fork the repo and make those changes [Look Here](https://github.com/tekartik/sqflite/compare/master...TinyProgrammers:master) and override_deps in pubspec.yaml

**Comment by matpascaud** [comment]
I am getting the same issue.

**Comment by Saifallak** [comment]
@matpascaud  yse my work around for now 

> add this to pubspec.yaml, and hola it works
> 
> ```yaml
> dependency_overrides:
>   sqflite:
>     git:
>       url: https://github.com/TinyProgrammers/sqflite.git
>       path: sqflite 
> ```

**Comment by matpascaud** [comment]
@Saifallak But then I am having another issue with FMDB
```
CocoaPods could not find compatible versions for pod "FMDB":
  In snapshot (Podfile.lock):
    FMDB (= 2.7.5, >= 2.7.5)

  In Podfile:
    sqflite (from `.symlinks/plugins/sqflite/ios`) was resolved to 0.0.3, which depends on
      FMDB (= 2.7.2)
```

**Comment by Plumillon** [comment]
Same issue here when trying to Archive to arm64.
I'm using sqflite through a dependency (floor : https://github.com/vitusortner/floor), I needed to override to use 1.3.0 to be able to build :
`dependency_overrides:
  sqflite: 1.3.0`

**Comment by egodefroy** [comment]
> had to fork the repo and make those changes [Look Here](https://github.com/tekartik/sqflite/compare/master...TinyProgrammers:master) and override_deps in pubspec.yaml
> 
> add this to pubspec.yaml, and hola it works
> 
> ```yaml
> dependency_overrides:
>   sqflite:
>     git:
>       url: https://github.com/TinyProgrammers/sqflite.git
>       path: sqflite 
> ```

does not work for me

**Comment by leidig54** [comment]
This happened to me when I changed the deployment target in Xcode from iOS 9.0 to 13.0. Reverting back to 9.0 fixed the issue for me.

**Comment by jacobokoenig** [comment]
Running on a real device works for me fsr

**Comment by egodefroy** [comment]
@jacobokoenig
Le problem is when building the release (flutter build ios)

**Comment by Pcol10** [comment]
> > had to fork the repo and make those changes [Look Here](https://github.com/tekartik/sqflite/compare/master...TinyProgrammers:master) and override_deps in pubspec.yaml
> > add this to pubspec.yaml, and hola it works
> > ```yaml
> > dependency_overrides:
> >   sqflite:
> >     git:
> >       url: https://github.com/TinyProgrammers/sqflite.git
> >       path: sqflite 
> > ```
> 
> does not work for me

Till new update won't appear use this 

```
dependency_overrides:
  sqflite: 1.3.1+2
```

**Comment by alextekartik** [comment]
So far I cannot plan any update since...I cannot reproduce:
- does it happen for everyone when creating a new project?
- can someone share his failing project so I can try to reproduce (and fix)?

I think (but I'm not sure) that it mainly happens for old projects and I guess some settings must be change in Xcode. Also has anyone be able to solve the issue (besides reverting to an old version) and can share his tips? Thanks!

**Comment by egodefroy** [comment]
@alextekartik

I'm on channel 1.24.0-8.0.pre.129

I created a new project from VSCode (1.51.1) and just change the first line in Podfile to:  platform :ios, '11.0'  and changed the bundle id to be able to build the release (flutter build ios) with XCode 12.2 and I got the following error without even referring to sqflite in the dart code.

I think you should be able to reproduce this.

````
Running pod install...                                           1 870ms
Running Xcode build...                                                  
Xcode build done.                                           17,4s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **


Xcode's output:
↳
    ld: warning: ignoring file
    /Users/egylis/Documents/Development/Flutter/testsqflite/build/ios/Release-iphoneos/FMD
    B/FMDB.framework/FMDB, building for iOS-armv7 but attempting to link with file built
    for iOS-arm64
    Undefined symbols for architecture armv7:
      "_OBJC_CLASS_$_FMDatabaseQueue", referenced from:
          objc-class-ref in SqflitePlugin.o
    ld: symbol(s) not found for architecture armv7
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description

````

**Comment by alextekartik** [comment]
Thanks @egodefroy for the steps.

However even by following your steps (on flutter beta), it works for me... I'm sorry, I'm definitely not an expert in iOS build so if someone has some clue, that would be helpful...

**Comment by vanfresh** [comment]
Not 100% sure, but I had to upgrade the deployment target to 10.0 to use another library. 

Step)
1) platform :ios, '10.0'
2) open XCode and set the deployment target to the same version there

This seems to have solved it for me.

**Comment by vanfresh** [comment]
Arm7s is for [iPhone 3-5](https://medium.com/@edisonlo/ios-framework-file-was-built-for-x86-64-which-is-not-the-architecture-being-linked-arm64-linker-4d3780669aa3).   

Hence it doesn't build if the minimum is iOS 9.0 (older iPhones doesn't support iOS 9.0).  I fixed it by disabling arm7s builds in **Architectures** under Build Settings.  You can do this by setting it as **arm64** rather than the $(ARCHS_STANDARD) which is both armv7s and arm64. 

![Screen Shot 2020-11-27 at 2 56 00 PM](https://user-images.githubusercontent.com/68772329/100420223-04115c00-30c1-11eb-9063-805eea766b20.png)

**Comment by dpatel** [comment]
> @jacobokoenig
> Le problem is when building the release (flutter build ios)

I am getting the same issue any solution?

**Comment by GyuriMajercsik** [comment]
> Arm7s is for [iPhone 3-5](https://medium.com/@edisonlo/ios-framework-file-was-built-for-x86-64-which-is-not-the-architecture-being-linked-arm64-linker-4d3780669aa3).
> 
> Hence it doesn't build if the minimum is iOS 11.0 (I think this version onward doesn't support arm7s). I fixed it by disabling arm7s builds in **Architectures** under Build Settings. You can do this by setting it as **arm64** rather than the $(ARCHS_STANDARD) which is both armv7s and arm64.
> 
> ![Screen Shot 2020-11-27 at 2 56 00 PM](https://user-images.githubusercontent.com/68772329/100420223-04115c00-30c1-11eb-9063-805eea766b20.png)

Same issue here but this solved for me.

**Comment by ahmtshn** [comment]
Try this. This solved for me. Change pods architecrures

<img width="1080" alt="Screen Shot 2021-01-02 at 01 28 55" src="https://user-images.githubusercontent.com/32811582/103447267-e2942900-4c99-11eb-95d5-c240eb7183ad.png">

**Comment by fryette** [comment]
> had to fork the repo and make those changes [Look Here](https://github.com/tekartik/sqflite/compare/master...TinyProgrammers:master) and override_deps in pubspec.yaml
> 
> add this to pubspec.yaml, and hola it works
> 
> ```yaml
> dependency_overrides:
>   sqflite:
>     git:
>       url: https://github.com/TinyProgrammers/sqflite.git
>       path: sqflite 
> ```

Can confirm that this fixes all issues
Workarounds with changing XCode pods settings not working for me

**Comment by alextekartik** [comment]
If someone could share a failing project, I could try to find the steps or fix the issue.

**Comment by fryette** [comment]
@alextekartik got the same issue with another project.
Seems when you trying to increase iOS version to 11 and then build a release it's failed

**Comment by fryette** [comment]
@alextekartik had this issue again with another project.
Seems when you set minimum iOS version to 11.0 including all IPHONEOS_DEPLOYMENT_TARGET, the release build failed.

**Comment by xqdad** [comment]
移除armv7，只保留arm64，解决问题

**Comment by alextekartik** [comment]
I encounter a similar issue when using bluetooth. One solution I found was to add the following at the end of the Podfile to force iOS 11. People still encountering this issue might try this:

```
post_install do |installer|
  installer.pods_project.targets.each do |target|
    flutter_additional_ios_build_settings(target)
    target.build_configurations.each do |config|
      config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '11.0'
    end
  end
end
```

**Comment by egorikem** [comment]
> 移除armv7，只保留arm64，解决问题
Why would you use Chinese in an English thread? Just why...

**Comment by trezm** [comment]
I tried the override with `sqflite: ^2.0.0+3`, but get the same issue. Anyone else successfully using a non-nullable version?

**Comment by Puruchandra** [comment]
This issue came when i set ios platform version from 9.0 to 11.0
I solved the issue by upgrading my sqflite dependency in pubspec.yaml

`dependencies:
  flutter:
    sdk: flutter
  sqflite: 1.3.1+2
`

**Comment by DFelten** [comment]
It's working for us to exlclude armv7 in the `Podfile` as `post_install`:

```ruby
post_install do |installer|
  installer.pods_project.build_configurations.each do |config|
    config.build_settings["EXCLUDED_ARCHS"] = "armv7"
  end
end
```

**Comment by taskcrypto** [comment]
I use this package with changed network image.
`post_install do |installer|
  installer.pods_project.targets.each do |target|
    flutter_additional_ios_build_settings(target)
  end
  installer.pods_project.build_configurations.each do |config|
      config.build_settings["EXCLUDED_ARCHS"] = "armv7"
  end
end`

**Comment by alextekartik** [comment]
Thanks @DFelten that seems to be the correct solution. Unfortunately I don't see how I can include this fix in sqflite itself. So far the solution has to be done in the main project. I have added some notes about this here: https://github.com/tekartik/sqflite/blob/master/sqflite/doc/troubleshooting.md#undefined-symbols-for-architecture-armv7

**Comment by nijatmursali** [comment]
None of them solved my issue. Any fix on M1 Air?

**Comment by SeanCodeMedia** [comment]
> It's working for us to exlclude armv7 in the `Podfile` as `post_install`:
> 
> ```ruby
> post_install do |installer|
>   installer.pods_project.build_configurations.each do |config|
>     config.build_settings["EXCLUDED_ARCHS"] = "armv7"
>   end
> end
> ```

seem to work for me. I added this line to my pod file and then delete my sqlite folder. Then I do pod install seem to fixed it

**Comment by irfan-429** [comment]
> This issue came when i set ios platform version from 9.0 to 11.0
> I solved the issue by upgrading my sqflite dependency in pubspec.yaml
> 
> `dependencies: flutter: sdk: flutter sqflite: 1.3.1+2 `

and what for them who are not using sqflite in project?

**Comment by irfan-429** [comment]
> 7\. pod install

I tried each solution provided above. Except this one none worked for me. Thanks for great solution :)

**Comment by chinloyal** [comment]
> Try this. Change pods architectures
> 
> <img alt="Screen Shot 2021-01-02 at 01 28 55" width="1080" src="https://user-images.githubusercontent.com/32811582/103447267-e2942900-4c99-11eb-95d5-c240eb7183ad.png">

This worked for me
